common/src/main/java/gov/cms/ab2d/common/health/UrlAvailable.java

```diff
@@ -0,0 +1,53 @@
+package gov.cms.ab2d.common.health;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.Socket;
+import java.net.URL;
+import java.util.List;
+
+@Slf4j
+/**
+ * Does health checks on URLs
+ */
+public class UrlAvailable {
+    /**
+     * Given a URL, test to see if you can open a socket to that domain on the specified or default port
+     *
+     * @param testUrl - The URL to test
+     * @return true if you can open a socket to that port on that server
+     */
+    public static boolean available(String testUrl) {
+        try {
+            URL url = new URL(testUrl);
+            int port = url.getPort() == -1 ? url.getDefaultPort() : url.getPort();
+            try (Socket socket = new Socket(url.getHost(), port)) {
+                log.debug("Socket opened to " + testUrl + " and is connected: " + socket.isConnected());
+            } catch (IOException e) {
+                log.info("Unable to open socket to " + testUrl);
+                return false;
+            }
+            return true;
+        } catch (MalformedURLException e1) {
+            log.info("Malformed URL: " + testUrl);
+            return false;
+        }
```

Comment:
A shortcut: https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#toString(java.net.URL,%20java.nio.charset.Charset)

Reply:
I just want to check to see if I can connect on the necessary port. I don't want the contents. That seems like a more expensive operation.

Reply:
👍